backend/domain_events/handler.py

```diff
@@ -41,6 +44,128 @@ def get_name(user_data, fallback: str = "<no name specified>"):
     )
 
 
+def handle_grant_reply_approved_sent(data):
+    is_reminder = data["is_reminder"]
+    grant = Grant.objects.get(id=data["grant_id"])
+    reply_url = urljoin(settings.FRONTEND_URL, "/grants/reply/")
+
+    subject = (
+        "Reminder: Financial Aid Update" if is_reminder else "Financial Aid Update"
+    )
+
+    template = None
+    variables = {
+        "replyLink": reply_url,
+        "startDate": f"{grant.conference.start:%-d %B}",
+        "endDate": f"{grant.conference.end:%-d %B}",
+        "deadlineDateTime": f"{grant.applicant_reply_deadline:%-d %B %Y %H:%M %Z}",
+        "deadlineDate": f"{grant.applicant_reply_deadline:%-d %B %Y}",
+    }
+    if grant.approved_type == Grant.ApprovedType.ticket_only:
+        template = EmailTemplate.GRANT_APPROVED_TICKET_ONLY
+    elif grant.approved_type == Grant.ApprovedType.ticket_travel_accommodation:
+        template = EmailTemplate.GRANT_APPROVED_TICKET_TRAVEL_ACCOMMODATION
+        variables["amount"] = f"{grant.approved_amount:.0f}"
+    else:
+        raise ValueError("Grant Approved type must be not null.")
+
+    _grant_send_email(template=template, subject=subject, grant=grant, **variables)
+
+    grant.status = Grant.Status.waiting_for_confirmation
+    grant.save()
+
+
+def handle_grant_reply_waiting_list_sent(data):
+    grant = Grant.objects.get(id=data["grant_id"])
+    reply_url = urljoin(settings.FRONTEND_URL, "/grants/reply/")
+
+    subject = "Financial Aid Update"
+    deadline = grant.conference.deadlines.filter(
+        type="custom", name__contains={"en": "Update Grants in Waiting List"}
+    ).first()
+
+    _grant_send_email(
+        template=EmailTemplate.GRANT_WAITING_LIST,
+        subject=subject,
+        grant=grant,
+        replyLink=reply_url,
+        grantsUpdateDeadline=f"{deadline.start:%-d %B %Y}",
+    )
+
+
+def handle_grant_reply_rejected_sent(data):
+    grant = Grant.objects.get(id=data["grant_id"])
+
+    subject = "Financial Aid Update"
+
+    _grant_send_email(
+        template=EmailTemplate.GRANT_REJECTED,
+        subject=subject,
+        grant=grant,
+    )
+
+
+def _grant_send_email(template: EmailTemplate, subject: str, grant: Grant, **kwargs):
+    try:
+        users_result = execute_service_client_query(
+            USERS_NAMES_FROM_IDS, {"ids": [grant.user_id]}
+        )
+
+        user_data = users_result.data["usersByIds"][0]
```

Comment:
don't we have a way to the data of a single user? we can leave it as it is for now but not ideal